### PR TITLE
Updating IDL version for go client.

### PR DESCRIPTION
**Pull request by mindaugasbarcauskas**
**What changed?**

Updated idl version

**Why?**

**How did you test it?**

CI pipeline.

**Potential risks**

### CODE REVIEW

File: internal/compatibility/proto/request.go [hunk 1]
```diff
@@ -623,3 +623,16 @@ func RefreshWorkflowTasksRequest(r *shared.RefreshWorkflowTasksRequest) *apiv1.R
...
+
+func RestartWorkflowExecutionRequest(r *shared.RestartWorkflowExecutionRequest) *apiv1.RestartWorkflowExecutionRequest {
+	if r == nil {
+		return nil
+	}
+	request := apiv1.RestartWorkflowExecutionRequest{
+		Domain:            r.GetDomain(),
+		WorkflowExecution: WorkflowExecution(r.GetWorkflowExecution()),
+		Identity:          r.GetIdentity(),
+	}
```
Comment (Groxx) [inline internal/compatibility/proto/request.go:635]: seems like proto is missing the `reason` field?  is that worth adding in now so it happens all at once?

Comment (Groxx) [reply internal/compatibility/proto/request.go:635]: ah, no, this proto API is missing a field that's in the thrift API.  we should fix that.

Comment (Groxx) [reply internal/compatibility/proto/request.go:635]: Tackling this in https://github.com/uber/cadence-idl/pull/128

Comment (mindaugasbarcauskas) [reply internal/compatibility/proto/request.go]: thanks Steven, very nice catch! I updated to latest idl where you included the reason and updated the mapper.

**Review by Groxx — APPROVED** [review]
LGTM 👍 

Probably worth choosing a more useful commit message though (e.g. it adds restart, it's not just a minor internal addition).

**Pull request merged**